landlab/components/soil_grading/soil_grading.py

```diff
@@ -0,0 +1,540 @@
+            )
+
+            self.g_state0 = grains_weight__distribution
+            self._grid.at_node["grains__weight"][self._grid.core_nodes, :] = 1
+            self._grid.at_node["grains__weight"] *= grains_weight__distribution
+            layer_depth = np.sum(self.g_state0) / (
+                self._soil_density * self._grid.dx * self._grid.dx
+            )
+            layer_depth /= 1 - self._phi
+
+            self._grid.at_node["soil__depth"][self._grid.core_nodes] += layer_depth
+            self._grid.at_node["topographic__elevation"] = (
+                self._grid.at_node["soil__depth"]
+                + self._grid.at_node["bedrock__elevation"]
+            )
+
+            self.g_state_bedrock = grains_weight__distribution
+            self._grid.at_node["bed_grains__proportions"][self._grid.core_nodes] = 1
+            self._grid.at_node["bed_grains__proportions"][
+                self._grid.core_nodes
+            ] *= np.divide(self.g_state_bedrock, np.sum(self.g_state_bedrock))
+
+        else:
+            self.g_state_bedrock = grains_weight__distribution
+            self._grid.at_node["bed_grains__proportions"][self._grid.core_nodes] = 1
+            self._grid.at_node["bed_grains__proportions"][
+                self._grid.core_nodes
+            ] *= np.divide(self.g_state_bedrock, np.sum(self.g_state_bedrock))
+
+    def _generate_normal_distribution(
+        self, median_size=None, std=None, total_soil_weight=None
+    ):
+
+        if median_size is None:
+            median_size = self._initial_median_size
+        if std is None:
+            std = self._CV * median_size
+        if total_soil_weight is None:
+            total_soil_weight = self._initial_total_soil_weight
+
+        lower = np.min(self._lowerlims)
+        upper = np.max(self._upperlims)
+
+        values = []
+        if median_size < lower:
+            grains_weight__distribution = np.zeros_like(self._upperlims)
+            grains_weight__distribution[0] = total_soil_weight
+            warnings.warn(
+                "Median size provided is smaller than the distribution lower bound",
+                stacklevel=2,
+            )
+
+        elif median_size > upper:
+            grains_weight__distribution = np.zeros_like(self._upperlims)
+            grains_weight__distribution[-1] = total_soil_weight
+            warnings.warn(
+                "Median size provided is larger than the distribution upper bound",
+                stacklevel=2,
+            )
+
+        else:
+            while np.size(values) < total_soil_weight:
+                sample = random.gauss(median_size, std)
+                if sample >= lower and sample <= upper:
+                    values.append(sample)
+
+            grains_weight__distribution = np.histogram(
+                values, np.insert(self._upperlims, 0, 0)
+            )[0]
+
+        return grains_weight__distribution
+
+    def update_median__size(self):
```

Comment:
Rename without the double underscore and, while you're at it, maybe rename it to `update_median_grain_size`?